hub/client/client.py

```diff
@@ -0,0 +1,122 @@
+import requests
+import sys
+from hub.client.config import (
+    HUB_REST_ENDPOINT,
+    GET_TOKEN_SUFFIX,
+    REGISTER_USER_SUFFIX,
+    DEFAULT_REQUEST_TIMEOUT,
+)
+from hub.client.utils import get_auth_header, check_response_status
+from hub.util.exceptions import LoginException
+from typing import Optional
+from hub import __version__
+
+
+class HubBackendClient:
+    """Communicates with Activeloop Backend"""
+
+    def __init__(self):
+        self.auth_header = None
+
+    def request(
+        self,
+        method: str,
+        relative_url: str,
+        endpoint: Optional[str] = None,
+        params: Optional[dict] = None,
+        data: Optional[dict] = None,
+        files: Optional[dict] = None,
+        json: Optional[dict] = None,
+        headers: Optional[dict] = None,
+        timeout: Optional[int] = DEFAULT_REQUEST_TIMEOUT,
+    ):
+        """Sends a request to the backend.
+
+        Args:
+            method (str): The method for sending the request. Should be one of 'GET', 'OPTIONS', 'HEAD', 'POST', 'PUT', 'PATCH', or 'DELETE'.
+            relative_url (str): The suffix to be appended to the end of the endpoint url.
+            endpoint(str, optional): The endpoint to send the request to.
+            params (dict, optional): Dictionary to send in the query string for the request.
+            data (dict, optional): Dictionary to send in the body of the request.
+            files (dict, optional): Dictionary of 'name': file-like-objects (or {'name': file-tuple}) for multipart encoding upload.
+                file-tuple can be a 2-tuple (filename, fileobj), 3-tuple (filename, fileobj, content_type)
+                or a 4-tuple (filename, fileobj, content_type, custom_headers), where 'content-type' is a string
+                defining the content type of the given file and 'custom_headers' a dict-like object containing 
+                additional headers to add for the file.
+            json (dict, optional): A JSON serializable Python object to send in the body of the request.
+            headers (dict, optional): Dictionary of HTTP Headers to send with the request.
+            timeout (float,optional): How many seconds to wait for the server to send data before giving up.
+
+        Returns:
+            requests.Response: The response received from the server.
+        """
+        params = params or {}
```

Comment:
Why not just set the defaults to these instead of None?

Reply:
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
Makes sense, thanks. Doesn't look like it applies here, but looks like a good general practice anyway.